packages/core/src/model.js

```diff
@@ -833,7 +833,7 @@ ${associationOwner._getAssociationDebugList()}`);
     const missingIndexes = this.getIndexes()
       .filter(item1 => !existingIndexes.some(item2 => item1.name === item2.name))
       .sort((index1, index2) => {
-        if (this.sequelize.options.dialect === 'postgres') {
+        if (['postgres', 'cockroachdb'].includes(this.sequelize.options.dialect)) {
```

Comment:
avoid creating throwaway arrays in production code. Tests are fine but this sort is going to be expensive if each iteration creates an array